### PR TITLE
processes: refactor get_process_by_listening_port

### PR DESCRIPTION
**Pull request by yotamolenik**

### CODE REVIEW

File: src/rpcclient/rpcclient/darwin/processes.py [hunk 1]
```diff
@@ -764,8 +764,9 @@ def grep(self, name: str) -> List[Process]:
                 result.append(p)
         return result
 
-    def get_process_by_listening_port(self, port: int) -> Optional[Process]:
+    def get_process_by_listening_port(self, port: int) -> List[Process]:
```
Comment (doronz88) [inline src/rpcclient/rpcclient/darwin/processes.py:767]: `get_processes_by_listening_port`

**Review by doronz88 — CHANGES_REQUESTED** [review]

File: src/rpcclient/rpcclient/darwin/processes.py (hunk 1 above)
Comment (yotamolenik) [reply src/rpcclient/rpcclient/darwin/processes.py]: done

Comment (doronz88) [reply src/rpcclient/rpcclient/darwin/processes.py]: Fix commit message accordingly

Comment (yotamolenik) [reply src/rpcclient/rpcclient/darwin/processes.py]: done

**Review by doronz88 — APPROVED** [review]

**Pull request merged**